messaging/messaging.go

```diff
@@ -913,13 +914,14 @@ func NewClient(ctx context.Context, c *internal.MessagingConfig) (*Client, error
 		return nil, errors.New("project ID is required to access Firebase Cloud Messaging client")
 	}
 
-	hc, _, err := transport.NewHTTPClient(ctx, c.Opts...)
+	c.Opts = append(c.Opts, internaloption.WithDefaultEndpoint(messagingEndpoint))
```

Comment:
Let's implement this without bringing in yet another package.

```
hc, endpoint, err := transport.NewHTTPClient(ctx, c.Opts...)
if err != nil {
		return nil, err
}

if endpoint == "" {
  endpoint = messagingEndpoint
}
```

Reply:
done :) also I've added tests